invenio_communities/communities/services/components.py

```diff
@@ -79,6 +79,18 @@ def create(self, identity, data=None, record=None, **kwargs):
     def update(self, identity, data=None, record=None, **kwargs):
         """Update handler."""
         self._populate_access_and_validate(identity, data, record, **kwargs)
+        visibility = data["access"]["visibility"]
+        opposite = "public" if visibility == "public" else "restricted"
```

Comment:
I do not like hardcoding this here, maybe could be delegated to the access sysfield, although `record.access.opposite_visibility`sounds quite ugly

Reply:
See the implementation I linked in the other comment: I ended up adding an `assert` to make sure.